lib/std/Build.zig

```diff
@@ -1525,6 +1525,21 @@ pub fn step(b: *Build, name: []const u8, description: []const u8) *Step {
     return &step_info.step;
 }
 
+pub fn standardLinkageOption(b: *Build, target: Target) std.builtin.LinkMode {
+    if (b.option(std.builtin.LinkMode, "linkage", "Method of linking binaries")) |linkage| {
+        return linkage;
+    }
+
+    if (target.abi.isGnu() or target.abi.isAndroid()) {
+        return .dynamic;
+    }
+
+    return switch (target.os.tag) {
+        .macos, .ios, .windows => .dynamic,
+        else => .static,
+    };
```

Comment:
```suggestion
pub fn standardLinkageOption(b: *Build, target: Target) ?std.builtin.LinkMode {
    return b.option(std.builtin.LinkMode, "linkage", "Method of linking binaries");
```

imo if this gets in it should only be providing the option with a standard name so that way there isnt multiple sources of truth for what this default is, given the logic for this is already in the compiler itself for handling when you don't pass it at all

Reply:
I was thinking it might be good to move that logic into `std.Target` so the compiler and builder could both use it.